tools/proofers/dp_proof.js

```diff
@@ -230,3 +230,17 @@ function transformText(transformType) {
     txtArea.focus();
 }
 
+// Avoid submitting twice if user double-clicks button
+// Usage: onClick="return submitJustOnce(this);"
+//
+// Uses a data attribute to store when button has been clicked for the first time
+// Returns true on first click to continue submission
+// If attribute is already set (second click), return false to avoid re-submitting
+function submitJustOnce(btn) {
```

Comment:
I would prefer we instead just mark the button disabled. That way for slow connections, there is an indication that something happened. Then we don't need to track anything since the 2nd click won't fire any event.

Reply:
That's what I tried first :)
However, once you disable the button, the submission doesn't happen, even when you return true.

I had an intermediate version that changed the button label, and used that as the flag detected by the second click. 
I decided the hidden data attribute was better because
1. It doesn't need you to choose suitable text to change the label to (though "Submitting" would probably be suitable for most.
2. The new label text doesn't need translating
3. It looks a bit odd: even when you're on a fast connection, you see the button label text change, which is unusual behavior.
4. This code seems fairly safe for anything that might not support the `dataset`. The double click bug would still be able to happen of course, but I don't think it should break normal single clicking.

Reply:
I agree.

Reply:
I just think that if clicking the button does nothing, it should be disabled. I tried this and it seems to achieve what you want. We don't need to change the label in my opinion.
https://jsfiddle.net/yo42jueb/
I'm not completely opposed to what we have here, just think we can use browser features (disabled button) instead of implementing something ourselves.

Reply:
Thanks @chrismiceli  - It may well be my naivety regarding the interaction between PHP, HTML and JS, and hence where to put everything, so I may have made the wrong edit, but my attempt to do what you suggested does not fix the bug, i.e. I still get the error if I double click. If I have done it correctly, I think the reason could be that by the time it's submitted, it's too late - the second click has already been made, which is why I had to use `onclick`, rather than `onsubmit`.  Maybe you can spot if I did something wrong in trying to copy the code from your jsfiddle:
I only tried it out on the WordCheck button, by commenting out the `onclick` I had previously added to `button_defs.inc`, adding your `submitForm` function to `dp_proof.js`, and adding `'onsubmit'="return submitForm()"` to the `<form...>` in `text_frame_std.inc`.
I'll try pushing a commit with those changes so you can see them in context and hopefully spot whether it's my error, or just that it doesn't trap the double click early enough.

Reply:
> Thanks @chrismiceli - It may well be my naivety regarding the interaction between PHP, HTML and JS, and hence where to put everything, so I may have made the wrong edit, but my attempt to do what you suggested does not fix the bug, i.e. I still get the error if I double click. If I have done it correctly, I think the reason could be that by the time it's submitted, it's too late - the second click has already been made, which is why I had to use `onclick`, rather than `onsubmit`. Maybe you can spot if I did something wrong in trying to copy the code from your jsfiddle: I only tried it out on the WordCheck button, by commenting out the `onclick` I had previously added to `button_defs.inc`, adding your `submitForm` function to `dp_proof.js`, and adding `'onsubmit'="return submitForm()"` to the `<form...>` in `text_frame_std.inc`. I'll try pushing a commit with those changes so you can see them in context and hopefully spot whether it's my error, or just that it doesn't trap the double click early enough.

@chrismiceli -- I can confirm that the code currently in @windymilla's sandbox can still be made to raise the error associated by double-clicking on the WordCheck button.

Reply:
> I've sent a PR to your branch @windymilla here: [windymilla#1](https://github.com/windymilla/dproofreaders/pull/1) If you are ok with that, you can merge it and it should show up here. Sandbox here: https://www.pgdp.org/~chrismiceli/c.branch/form/

I've tested in @chrismiceli's sandbox, and can confirm that it (as far as I can tell) fixes the issue.

Something oddball that I experienced -- the project that I chose for my testing had French for the language. Now, on TEST, this doesn't mean a whole lot, and as the pages were actually in English, I eventually got tired of adding English as a dictionary to WC for the times it came up successfully. So I changed it to English. Whereupon I couldn't get the double-click to error, even in FF 52. (I tested in FF 52, and Chrome 70 and 90.)

It's actually pretty irrelevant, because regardless, it's a good idea to fix the issue. I mention it only because it's a good example of where speed (i.e. loading dictionaries) does seem to make a difference.